cargo-dist/src/backend/ci/mod.rs

```diff
@@ -21,66 +25,133 @@ pub struct CiInfo {
     pub github: Option<GithubCiInfo>,
 }
 
-/// Get the command to invoke to install dist via sh script
-fn install_dist_sh_for_version(version: &Version) -> String {
-    if let Some(git) = install_dist_git(version) {
-        return git;
-    }
-    let format = cargo_dist_schema::format_of_version(version);
-    let installer_name = if format.unsupported() {
-        // FIXME: we should probably do this check way higher up and produce a proper err...
-        panic!("requested dist v{version}, which is not supported by the this copy of dist ({SELF_DIST_VERSION})");
-    } else if format.artifact_names_contain_versions() {
-        format!("cargo-dist-v{version}-installer.sh")
-    } else {
-        "cargo-dist-installer.sh".to_owned()
-    };
-
-    // FIXME: it would be nice if these values were somehow using all the machinery
-    // to compute these values for packages we build *BUT* it's messy and not that important
-    let installer_url = format!("{BASE_DIST_FETCH_URL}/v{version}/{installer_name}");
-    format!("curl --proto '=https' --tlsv1.2 -LsSf {installer_url} | sh")
+/// Gives us the full information re: the version of dist we're supposed
+/// to install/run in CI
+struct DistInstallSettings<'a> {
+    version: &'a Version,
+    url_override: Option<&'a CargoDistUrlOverrideRef>,
 }
 
-/// Get the command to invoke to install dist via ps1 script
-fn install_dist_ps1_for_version(version: &Version) -> String {
-    if let Some(git) = install_dist_git(version) {
-        return git;
+/// Generates github steps to install a tool
+pub trait InstallStrategy {
```

Comment:
Nice call on making this a trait!

Reply:
Thanks! It wasn't part of the original plan but the cargo auditable stuff happened!